### DISCUSSION TITLE
How to Create Pull Request / Commit

### DISCUSSION THREAD

**Discussion by AteroConfigs** [hub4j/github-api#1152] [Q&A] [answered]
I would like to know how to make a pull request or commit, with a String of content as comment. Is that possible or am i not quite understanding the possibility's of this API?
Thanks in advance

**Comment by akashRindhe** [comment, marked as answer]
A pull request and a commit are quite different from each other, so it looks like you have two separate questions. I'm addressing the pull request question below - 

Given you have an existing GitHub repository with a base branch `main` and a `new-feature` branch with commits, you can create a pull request to merge your `new-feature` branch into your `main` branch with the following code -

```java
GHRepository repo = gitHub.getOrganization("AteroConfigs").getRepository("repo-name-here");
String prTitle = "Adding new feature ABC";
String prDescriptionInMarkdown = ## PR Description goes here"
GHPullRequest pullRequest = repo.createPullRequest(prTitle, "new-feature", "main", prDescriptionInMarkdown);
```

If you want to add additional comments to the pull request, you can do so with the `pullRequest` object above
```java
pullRequest.comment("My new comment");
```

**Reply by Verschwiegener** [reply]
Thank you very much! 
Now all i need is how to make a commit

**Reply by Verschwiegener** [reply]
I just noticed, that i made this Discussion with my other Account

**Reply by akashRindhe** [reply]
For creating a commit, checkout the `createContent()` function in `GHRepository` class. 

Below is an example of its usage -
https://github.com/hub4j/github-api/blob/d0912009dd84c55190f01034a59aa3f8ec62fdf5/src/test/java/org/kohsuke/github/GHPullRequestTest.java#L362-L368

You can refer to the [github docs](https://docs.github.com/en/rest/reference/repos#create-or-update-file-contents) for creating a new commit to understand the meaning of the parameters.

**Reply by Verschwiegener** [reply]
Okay thanks i try it when Im home

**Reply by Verschwiegener** [reply]
When i try this i always get This Error:
"message":"Not Found","documentation_url" with a different docummentaion uRL depending on the think i want to achive
